### ISSUE TITLE
Validation error in bindings

### ISSUE THREAD

**Issue by cbizon** [TranslatorSRI/reasoner-pydantic#30]
Hi @alongreyber I am trying to use reasoner-pydantic 2.1.2 in aragorn ranker.

I'm running into an issue when a TRAPI query comes into a function expecting a PDResponse as an input.  The input to this function already has results (i.e. bindings) e.g.

```
"results": [
            {
                "node_bindings": {
                    "n0": [
                        {
                            "id": "MONDO:0021187"
                        }
                    ],
```

WIth the new pydantic models this is causing 422 errors like this:

```
[{"loc":["body","message","results","__root__",0,"node_bindings","__root__","n1"],"msg":"unhashable type: \'list\'","type":"type_error"},{"loc":["body","message","results","__root__",1,"node_bindings","__root__","n1"],"msg":"unhashable type: \'list\'","type":"type_error"},{"loc":["body","message","results","__root__",2,"node_bindings","__root__","n1"],"msg":"unhashable type: \'list\'","type":"type_error"},{"loc":["body","message","results","__root__",3,"node_bindings","__root__","n1"],"msg":"unhashable type: \'list\'",
```

It seems to me that there's a mismatch between the list coming in and the HashableSet in the definition, but I'm far from sure about that.  Any thoughts?

**Comment by cbizon** [comment]
OK, I think I understand now what's going on, at least somewhat.    EdgeBindings in the TRAPI model have an "attributes" list, but NodeBindings don't.   One of our components likes to add attributes to NodeBindings.   That's technically allowed I think, because "extra" is True on NodeBindings.   But without the hints for pydantic that exist in the EdgeBindings model, it doesn't know what to do with the atttribute list that's showing up in NodeBindings.

What I'm not too sure about is what the right solution is... Adding attributes to nodebinding in Trapi and then pushing that down here would work, as would removing the attributes from our nodebindings in AC.   Is there some other way for the pydantic model to handle things cleanly without requiring a TRAPI change?

**Comment by cbizon** [comment]
FWIW, I made a PR on TRAPI: https://github.com/NCATSTranslator/ReasonerAPI/pull/312

I do wonder if it would be fair to add the attributes to NodeBindings in the pydantic models just to make the encoder happy, even if they are not technically part of TRAPI?

**Comment by alongreyber** [comment]
It looks like we're not handling extra fields correctly when they are given. I'm a bit worried about the performance hit from extra fields because to make this work we would need to parse all extra fields into our custom format (converting lists to HashableSequence recursively). 

I like the idea of just adding NodeBinding attributes, so I can open a PR for that.

**Comment by maximusunc** [comment]
#36 is going to add this parsing of all extra fields in the message. There are too many places (#35 for example) where services can add arbitrary fields in different places for us to include everything in the TRAPI schema.

**Comment by maximusunc** [comment]
This should now be fixed by #36
